packages/shylu/shylu_dd/bddc/src/shylu_KrylovSolver.h

```diff
@@ -457,14 +548,14 @@ template <class SX, class SM, class LO, class GO>
     char COMPZ = 'N'; // calculate eigenvalues only
     SX *Z(0), *WORK(0);
     int LDZ(1);
-    Teuchos::LAPACK<int, double> LAPACK;
+    Teuchos::LAPACK<int, SX> LAPACK;
     LAPACK.STEQR(COMPZ, numRows, D, E, Z, LDZ, WORK, &INFO);
-    assert (INFO == 0);
+    BDDC_TEST_FOR_EXCEPTION(INFO != 0, std::runtime_error, "STEQR error");
```

Comment:
Consider not just throwing, but collecting up some error info to propagate on the next reduction.  What if some processes throw and others don't?  I don't insist on this, since many other solvers don't do it, but it's something to think about.

Reply:
Yes, I'll think about how to do this better.

Reply:
This is hard, but just something to think about for the future.